samples/base/Session.cpp

```diff
@@ -362,13 +384,21 @@ namespace darwin {
             }
         }
 
+        std::size_t packet_size;
+
+        if (hdr->certitude_size > 1) {
+            packet_size = sizeof(darwin_filter_packet_t) + sizeof(unsigned int) * (hdr->certitude_size - 1);
+        } else {
+            packet_size = sizeof(darwin_filter_packet_t);
+        }
+
```

Comment:
see other comment

Reply:
Corrected and improved b3c45cf38eb225abc923993b3a3eb036feff74b5